Sources/Protocols/MessagesLayoutDelegate.swift

```diff
@@ -80,6 +80,17 @@ public protocol MessagesLayoutDelegate: AnyObject {
     /// - Note:
     ///   The default value returned by this method is zero.
     func messageBottomLabelHeight(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> CGFloat
+    
+    /// Cell size calculator for messages with MessageType.custom.
+    ///
+    /// - Parameters:
+    ///   - message: The custom message
+    ///   - indexPath: The `IndexPath` of the cell.
+    ///   - messagesCollectionView: The `MessagesCollectionView` in which this cell will be displayed.
+    ///
+    /// - Note:
+    ///   The default implementation will throw fatalError(). You must override this method if you are using messages with MessageType.custom.
+    func cellSizeCalculator(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> CellSizeCalculator
```

Comment:
Hey @RealBonus, this is an interesting PR. Thanks for taking the time to submit it 👍 I'm a little concerned that this method is called `cellSizeCalculator` but is only used for the `.custom` case of `MessageKind`. Maybe `customCellSizeCalculator` is a better name? Aside from that, I'll let some other team members review this

Reply:
Good point, agreed